### PR TITLE
Fix for Issue #3517 could site name show current page name

### PR DESCRIPTION
**Pull request by pnrich**
This is a fix for issue #3517. I used an ng-bind in the title which was tied to a $root variable. The variable changes whenever the state changes, and support is included for translations.

### CODE REVIEW

**Comment by pnrich** [comment]
Just realized I didn't include any names for the static pages. Adding that in my next commit.

EDIT: Apparently this isn't necessary. The changes I made only affect the dynamic pages, leaving the titles of the static pages (front, contact, etc.) unchanged.

File: common/locales/en/generic.json [hunk 1]
```diff
@@ -3,6 +3,28 @@
     "stringNotFound": "String '<%= string %>' not found.",
     "titleIndex": "Habitica | Your Life The Role Playing Game",
     "habitica": "Habitica",
+    
+    "tasksTitleSuffix": "Tasks",
```
Comment (crookedneighbor) [inline common/locales/en/generic.json]: This is looking great, but lets name space it with "title" first, so:

``` js
"titleTasks": "Tasks",
"titleAvatar": "Avatar",
"titleBackgrounds": "Backgrounds",
// ... etc ...
```

File: website/public/js/controllers/rootCtrl.js [hunk 2]
```diff
@@ -15,6 +15,9 @@ habitrpg.controller("RootCtrl", ['$scope', '$rootScope', '$location', 'User', '$
 
     $rootScope.$on('$stateChangeSuccess',
       function(event, toState, toParams, fromState, fromParams){
+        
+        $rootScope.title = $state.current.title;
```
Comment (crookedneighbor) [inline website/public/js/controllers/rootCtrl.js]: Let's make this `$rootScope.pageTitle` so it's more specific about what it is.

**Comment by crookedneighbor** [comment]
Other than a few comments, this is good to be merged. Ping me when you've made the changes. 

Sorry it took me so long to look at this. I think you did a great job.

File: website/public/js/app.js [hunk 3]
```diff
@@ -40,6 +40,7 @@ window.habitrpg = angular.module('habitrpg',
         .when('/options/groups/hall', '/options/groups/hall/heroes')
         .when('/options/inventory', '/options/inventory/drops')
         .when('/options/settings', '/options/settings/settings')
+        //.when('/tasks', {templateUrl: "partials/main.html", title: 'Tasks'})
```
Comment (crookedneighbor) [inline website/public/js/app.js]: This line should be removed.

**Comment by pnrich** [comment]
@crookedneighbor I've made the changes. Should I make a separate pull request, or just a commit to my own develop branch?

**Comment by crookedneighbor** [comment]
You can just commit it to your develop branch and push it. It will update here.

In the future, I recommend branching off so you can continue to work on multiple things.

**Comment by crookedneighbor** [comment]
I've squashed your commits and merged them: https://github.com/HabitRPG/habitrpg/commit/56a00efb198a609f7d54591338ff32193fd83391

Note: Your develop branch is now out of sync with Habit's develop branch. If you're comfortable with the git command line, you can easily fix this. If not, it's probably easiest to delete your fork and re-fork it.

This will count towards a contributor tier! What's your User ID? You can find it in [Settings > API](https://habitrpg.com/#/options/settings/api). Be sure to only copy the User ID and _not_ your API Token.

**Comment by pnrich** [comment]
Thanks! My User ID is 5914861f-d267-45c5-9cfb-f6be2dca4e12

Glad I could help!

**Comment by crookedneighbor** [comment]
Tier awarded! You can read more about [contributor tiers and rewards here](http://habitrpg.wikia.com/wiki/Contributor_Rewards).

Note that as your tier increases, it takes progressively more work to achieve the next one, so for the higher tiers, larger PRs or a few PRs are required. There's some information about that at Contributor Rewards, but feel free to ask here if you have questions!

**Pull request closed without merging**